### ISSUE TITLE
[EPIC][In-App Config] Configure trace data stores independent of install process

### ISSUE THREAD

**Issue by kdhamric** [kubeshop/tracetest#1612]
Currently, the Tracetest server install command walks a user through a series of questions to setup the connection to the trace data store. This includes asking what type of trace data store (Jaeger, OpenSearch, SignalFX, etc) and then questions related to each. If all are answered properly, the system will work. If not, you either need to reinstall or manually change the server config file.

We want to move the initial configuration to be allowed later in the process and to be 'changeable' without restarting the server. This would allow a new user to do a straightforward install with little thought or fact finding beforehand. When they launch the tool, they would be prompted to setup a connection to the backend trace data store. This would ease the user journey - they would successfully install Tracetest as it would have very little information needed for the install to work. When adding the datastore, we could allow them to enter it and test it, with a UI guiding the journey.

The trace data store wizard should:
  - ask the type of trace data store. The sequence of questions asked based on this will be different
  - ask the specific questions based on the data store (url, auth info, etc)
  - do a test against the trace data store
  
In the configuration screen you should be able to see the trace data store configuration, test the configuration, and alter the configuration.

### Tickets for this epic

- [x] #1638
- [x] #1625
- [x] #1622
- [x] #1627 
- [x] #1640
- [x] #1628
- [x] #1629
- [x] #1630
- [x] #1621 
- [x] #1654
- [x] #1656 
- [x] #1657 
- [x] #1679 
- [x] #1686 
- [x] #1687
- [x] #1688
- [x] #1706
- [x] #1693
- [x] #1703
- [x] #1700
- [x] #1702
- [x] #1704

### List of bugs/improvements

- [x] #1739
- [x] #1732 
- [x] #1751

**Comment by mathnogueira** [comment]
### Allow a trace data store to be configured via the CLI after install

I was thinking about enabling configs to be added/deleted with:

**Add**
```sh
tracetest server data-source apply -f data-source.yaml
```
```yaml
type: DataStore
spec:
  # just like environments, data store slugs would be named all lowercase with dashes (-) instead of spaces.
  # so it would become "production-jaeger"
  name: Production Jaeger
  type: jaeger
  jaeger:
    url: prod.jaeger.my-company.com
    insecure: true
```

**Delete**

```sh
tracetest server data-source delete <data-source-slug>
```

**Comment by xoscar** [comment]
In-App Configuration High Level Requirements:

Chunks:
1. Post-setup wizard
2. No-Tracing mode (pre-setup)
3. Installer Update

Post-setup Wizard Acceptance Criteria

AC1:
As a user installing Tracetest
I should be able to install the app
Without having to specify the tracing setup data (data store)

AC2:
As a user installing Tracetest
And I have decided to not provide the data store info at the beginning
And having just installed Tracetest
I should be able to see both the Server and UI working as expected

AC3:
As a user that just installed Tracetest
And I have decided to not provide the data store info at the beginning
While looking at the UI I should be able to recognize that the app is not fully ready to handle traces

AC4:
As a user that just installed Tracetest
And I have decided to not provide the data store info at the beginning
I should be able to start the setup wizard from the UI

AC5:
As a user looking at the setup wizard
I should be able to see a brief explanation about why a data store is needed
And a way to select the data store of my preference

AC6:
As a user looking at the UI setup wizard
And I select the Otel Collector as my data store
I should see a small tutorial with the steps I need to take to configure it

AC7:
As a user looking at the UI setup wizard
And I have selected the data store of my preference
I should be able to see the different options required to configure the data store

AC8:
As a user looking at the UI setup wizard
After filling in the details to configure the data store
I should be able to test the connection

AC9:
As a user looking at the UI setup wizard
After filling in the details to configure the data store
And I choose the test connection option
I should be able to see the result (error or success)
And details about the results

AC10:
As a user looking at the UI setup wizard
After filling in the details to configure the data store
And I have tested that the connection is successful
I should be able to save the changes

AC11:
As a user that just went through the setup process
I should see a change in the UI that will tell me that Tracetest is ready to support traces

AC12:
As a user looking at the Tracetest UI
I should be able to restart the setup process
And being able to update and save the settings

AC13:
As a user trying to update the setup setting
If there are some tests or transactions in progress
I should see a warning telling me about what is going to happen after the change

No-Tracing mode Acceptance Criteria

AC1:
As a user that just installed Tracetest
And I have decided to not provide the data store info at the beginning
I should be able to check if the app is in no-tracing mode

AC2:
As a user that just installed Tracetest
And I have decided to not provide the data store info at the beginning
I should be able to create tests

AC3:
As a user that just created a test in no-tracking mode
I should be redirected to the run page
And see the immediate result of the service

AC4:
As a user looking at a no-tracking mode test
I should be able to move to the trace and test modes
And the trace should display a single trace trigger span

AC5:
As a user looking at a no-tracking mode test
I should be able to create test specs on the trace trigger span

AC6:
As a user looking at a no-tracking mode test
And I move to either trace or test mode
I should be prompted with information to finalize the tracetest setup
And a way to initialize the setup that leads me to the wizard

Installer Update Acceptance Criteria

AC1:
As a user looking wanting to install Tracetest using the CLI
I should be able to start the process without having to get asked about my tracing setup
And both Server and UI should work as expected
